chain/chaintest/balance_handler.go

```diff
@@ -0,0 +1,191 @@
+		// Assertion is run after TestFunc for any additional checks
+		// State keys are not enforced in Assertion
+		Assertion func(context.Context, *testing.T, state.Mutable)
+
+		ExpectedBalance uint64
+		ExpectErr       bool
+	}{
+		{
+			Name:   "simpleAddBalance",
+			Actor:  addrOne,
+			Amount: 1,
+			Setup: func(ctx context.Context, t *testing.T, mu state.Mutable) {
+				r := require.New(t)
+				r.NoError(b.AddBalance(ctx, addrOne, mu, 0))
+			},
+			TestFunc:        b.AddBalance,
+			ExpectedBalance: 1,
+		},
+		{
+			Name:   "addBalanceFailsIfOverflow",
+			Actor:  addrOne,
+			Amount: 1,
+			Setup: func(ctx context.Context, t *testing.T, mu state.Mutable) {
+				// Initialize addrOne balance to MaxUint64
+				r := require.New(t)
+				r.NoError(b.AddBalance(ctx, addrOne, mu, consts.MaxUint64))
+			},
+			TestFunc:        b.AddBalance,
+			ExpectedBalance: consts.MaxUint64,
+			ExpectErr:       true,
+		},
+		{
+			Name:   "simpleDeduct",
+			Actor:  addrOne,
+			Amount: 1,
+			Setup: func(ctx context.Context, t *testing.T, mu state.Mutable) {
+				// Initialize addrOne balance to 1
+				r := require.New(t)
+				r.NoError(b.AddBalance(ctx, addrOne, mu, 1))
+			},
+			TestFunc:        b.Deduct,
+			ExpectedBalance: 0,
+		},
+		{
+			Name:   "deductFailsIfNotEnoughBalance",
+			Actor:  addrOne,
+			Amount: 2,
+			Setup: func(ctx context.Context, t *testing.T, mu state.Mutable) {
+				// Initialize addrOne balance to 1
+				r := require.New(t)
+				r.NoError(b.AddBalance(ctx, addrOne, mu, 1))
+			},
+			TestFunc:        b.Deduct,
+			ExpectedBalance: 1,
+			ExpectErr:       true,
+		},
+		{
+			Name:   "simpleCanDeduct",
+			Actor:  addrOne,
+			Amount: 1,
+			Setup: func(
+				ctx context.Context,
+				t *testing.T,
+				mu state.Mutable,
+			) {
+				// Initialize addrOne balance to 1
+				r := require.New(t)
+				r.NoError(b.AddBalance(ctx, addrOne, mu, 1))
+			},
+			TestFunc: func(
+				ctx context.Context,
+				addr codec.Address,
+				mu state.Mutable,
+				amount uint64,
+			) error {
+				return b.CanDeduct(ctx, addr, mu, amount)
+			},
+			ExpectedBalance: 1,
+		},
+		{
+			Name:   "canDeductFailsIfNotEnoughBalance",
+			Actor:  addrOne,
+			Amount: 2,
+			Setup: func(ctx context.Context, t *testing.T, mu state.Mutable) {
+				// Initialize addrOne balance to 1
+				r := require.New(t)
+				r.NoError(b.AddBalance(ctx, addrOne, mu, 1))
+			},
+			TestFunc: func(
+				ctx context.Context,
+				addr codec.Address,
+				mu state.Mutable,
+				amount uint64,
+			) error {
+				return b.CanDeduct(ctx, addr, mu, amount)
+			},
+			ExpectedBalance: 1,
+			ExpectErr:       true,
+		},
+		{
+			Name:   "multipleAccountsCanExist",
+			Actor:  addrTwo,
+			Amount: 2,
+			Setup: func(ctx context.Context, t *testing.T, mu state.Mutable) {
+				r := require.New(t)
+				// Initialize addrOne balance to 1
+				r.NoError(b.AddBalance(ctx, addrOne, mu, 1))
+				// Initialize addrTwo balance to 0
+				r.NoError(b.AddBalance(ctx, addrTwo, mu, 0))
```

Comment:
nit: Don't feel obligated to comment all of this - most of this code is pretty self-describing so the comments aren't really necessary IMO. I think either a high-level comment on the test or a descriptive test name is sufficient